packages/overlay/src/integrations/sentry/components/insights/agents/AITraceDetails.tsx

```diff
@@ -0,0 +1,345 @@
+import { useState } from "react";
+import { useParams } from "react-router-dom";
+import {
+  createAITraceFromSpan,
+  detectAILibraryHandler,
+} from "~/integrations/sentry/components/insights/agents/sdks/aiLibraries";
+import type { AILibraryHandler, SpotlightAITrace } from "~/integrations/sentry/types";
+import { getFormattedDuration } from "~/integrations/sentry/utils/duration";
+import { Badge } from "~/ui/badge";
+import SidePanel, { SidePanelHeader } from "~/ui/sidePanel";
+import Table from "~/ui/table";
+import { SearchProvider } from "../../../context/SearchContext";
+import useSentryStore from "../../../store";
+import DateTime from "../../shared/DateTime";
+import SpanTree from "../../traces/spans/SpanTree";
+
+const AGENTS_ROUTE = "/insights/agents";
+
+interface ToolCallDetailProps {
+  toolCall: {
+    toolCallId: string;
+    toolName: string;
+    args: Record<string, unknown>;
+    result?: Record<string, unknown> | string;
+    state?: string;
+    step?: number;
+  };
+}
+
+function ToolCallDetail({ toolCall }: ToolCallDetailProps) {
+  return (
+    <div className="border-primary-700 mb-4 rounded border p-4">
+      <h3 className="mb-2 font-bold">{toolCall.toolName}</h3>
+      <div className="text-primary-400 mb-2 text-sm">ID: {toolCall.toolCallId}</div>
+
+      <div className="mb-2">
+        <h4 className="text-sm font-semibold uppercase">Arguments</h4>
+        <div className="bg-primary-900 p-2">
+          <pre className="whitespace-pre-wrap break-words font-mono text-sm">
+            {JSON.stringify(toolCall.args, null, 2)}
+          </pre>
+        </div>
+      </div>
+
+      {toolCall.result && (
+        <div>
+          <h4 className="text-sm font-semibold uppercase">Result</h4>
+          <div className="bg-primary-900 p-2">
+            {typeof toolCall.result === "string" ? (
+              <pre className="whitespace-pre-wrap break-words font-mono text-sm">{toolCall.result}</pre>
+            ) : (
+              <pre className="whitespace-pre-wrap break-words font-mono text-sm">
+                {JSON.stringify(toolCall.result, null, 2)}
+              </pre>
+            )}
+          </div>
+        </div>
+      )}
+    </div>
+  );
+}
+
+function AITraceMetadata({ trace, handler }: { trace: SpotlightAITrace; handler: AILibraryHandler }) {
+  const metadata = [
+    ["Trace ID", trace.id],
+    ["Type", handler.getTypeBadge(trace)],
+    ["Operation", trace.operation],
+    ["Timestamp", <DateTime key="timestamp" date={trace.timestamp} />],
+    ["Duration", getFormattedDuration(trace.durationMs)],
+    ["Tokens (prompt/completion)", handler.getTokensDisplay(trace)],
+  ];
+
+  if (trace.metadata.functionId) {
+    metadata.push(["Function ID", trace.metadata.functionId]);
+  }
+
+  if (trace.metadata.maxRetries !== undefined) {
+    metadata.push(["Max Retries", String(trace.metadata.maxRetries)]);
```

Comment:
protip: Can also use template literals for string casting
```suggestion
    metadata.push(["Max Retries", `${trace.metadata.maxRetries}`]);
```